src/cgame/rocket/rocketChatField.h

```diff
@@ -116,27 +116,27 @@ class RocketChatField : public Rml::Element, Rml::EventListener
 					case Rml::Input::KI_BACK:
 						if ( cursor_character_index > 0 )
 						{
-							text.erase( cursor_character_index - 1, 1 );
+							MoveCursorBackward();
+							text.erase( cursor_character_index, Q_UTF8_Width( text.c_str() + cursor_character_index ) );
 							UpdateText();
-							MoveCursor( -1 );
 						}
 						break;
 
 					case Rml::Input::KI_DELETE:
-						if ( cursor_character_index < (int) text.size() )
+						if ( cursor_character_index < text.size() )
 						{
-							text.erase( cursor_character_index, 1 );
+							text.erase( cursor_character_index, Q_UTF8_Width( text.c_str() + cursor_character_index ) );
```

Comment:
How come backspace and delete have the same erase code? Shouldn't delete be deleting the character *in front* of the cursor while backspace deletes the character *behind* the cursor?

Reply:
Backspace is implemented like leftarrow then delete.